### PR TITLE
Version check

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
As suggested in https://github.com/orocos/orocos_kinematics_dynamics/issues/296#issuecomment-724181063

I am not sure we should bump to `1.5.0` or `2.0.0`

### CODE REVIEW

**Review by meyerj — COMMENTED** [review]
Thanks, @MatthijsBurgh, for picking this suggestion up so quickly. In https://github.com/orocos/orocos_kinematics_dynamics/issues/296#issuecomment-724181063 I did not mean to actually bump the KDL version to 1.5, only to add a version check. However, there are some other API-breaking open pull requests that might be worth to do that (e.g. https://github.com/orocos/orocos_kinematics_dynamics/pull/160) soon, but not necessarily for `python_orocos_kdl`.

But there have been some updates that break building the Python bindings within 1.4, like the addition of `KDL::Joint::Fixed` in  https://github.com/orocos/orocos_kinematics_dynamics/commit/dce6742a379698bc6d50afedc8c0a4fc4657d493 (https://github.com/orocos/orocos_kinematics_dynamics/pull/229). For C++ users that was fully backwards-compatible, but `python_orocos_kdl` requires a version that defines this enum constant since then.

Extending what I proposed earlier, I therefore suggest to integrate the patch version number when finding `orocos_kdl` in `python_orocos_kdl`:
```cmake
# find a matching version of orocos_kdl
find_package(orocos_kdl ${PROJECT_VERSION} EXACT REQUIRED)
```
and to bump both version numbers to 1.4.1 (and not yet to 1.5.0). Then at least `python_orocos_kdl` fails to find `orocos_kdl` if it is installed from the Debian/Ubuntu package [orocos-kdl](https://packages.debian.org/source/sid/orocos-kdl) where the upstream version 1.4.0 [has been imported two years ago](https://salsa.debian.org/science-team/orocos-kdl/-/commit/29c4dfe1eda4c5cc033d5040cd3953043ca0b7ec).

I am not even sure whether `EXACT` works with only the major.minor part. Maybe it would only match a version with patch number 0?

In the future we would then need to

- bump the minor version number for [ABI-breaking changes](https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B) (e.g. modify the signature of an existing C++ method)
- bump the patch version number for all changes that do not break the ABI, but require a synchronized update in `orocos_kdl` and `python_orocos_kdl` (e.g. add a new non-virtual member function to both).

@smits ?

**Review by smits — COMMENTED** [review]
These changes should not trigger a minor version bump afaiu

**Comment by smits** [comment]
I agree with what @meyerj says.

File: orocos_kdl/package.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
   schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>orocos_kdl</name>
-  <version>1.4.0</version>
+  <version>1.5.0</version>
```
Comment (meyerj) [inline orocos_kdl/package.xml:7]:
```suggestion
  <version>1.4.1</version>
```

File: python_orocos_kdl/package.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>python_orocos_kdl</name>
-  <version>1.4.0</version>
+  <version>1.5.0</version>
```
Comment (meyerj) [inline python_orocos_kdl/package.xml:7]:
```suggestion
  <version>1.4.1</version>
```

File: python_orocos_kdl/doc/conf.py [hunk 3]
```diff
@@ -38,16 +40,16 @@
 
 # General information about the project.
 project = u'Orocos KDL python bindings'
-copyright = u'2012, Ruben Smits'
+copyright = u'2020, Ruben Smits'
```
Comment (meyerj) [inline python_orocos_kdl/doc/conf.py:43]: I am not a lawyer, but I think the year in copyright statements is the year of the first publication (or a range of years, like 2012-2020). I would not update it to 2020 or later here and elsewhere.

Seems like many copyright statements have been updated in https://github.com/orocos/orocos_kinematics_dynamics/commit/61f457a8a759c8e1f8a491dbbc9ca97eebb886bf?

Reference:
- https://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated

Comment (MatthijsBurgh) [reply python_orocos_kdl/doc/conf.py]: I checked the entire repo for changes since the 1.4 release. I updated this in #298

**Review by meyerj — APPROVED** [review]
LGTM, thanks!

**Pull request merged**